test/code/plugins/oms_diag_lib_test.rb

```diff
@@ -240,4 +254,95 @@ def test_case_record_proc_01
                      "#{OMS::Diag::RECORD_MGID} key should be equal to #{OMS::Diag::RECORD_MGID_VALUE}")
     end
 
+    # Test cases pertaining to version check
+
+    # Checking diagnostic support with lower omsagent version
+    def test_case_version_check_01
+        versions = ["1.3.4-126", "1.3.3-127", "1.2.4-127", "0.3.4-127"]
+
+        versions.each do |v|
+            # Setting up version file
+            File.open(OMS::Diag.GetInstallInfoPath(), 'w') do |file|
+                file.puts "#{v} xcwer23234 Release_Build"
+            end
+
+            # Forcing parsing of version file
+            OMS::Diag.SetDiagSupported(nil)
+
+            # Checking the diag is not enabled
+            assert_equal(false,
+                     OMS::Diag.IsDiagEnabled(),
+                     "Diagnostic logging should not be supported for lower version #{v} than min version")
+        end
+    end
+
+    # Checking diagnostic support with min omsagent version
+    def test_case_version_check_02
+        # Setting up version file
+        File.open(OMS::Diag.GetInstallInfoPath(), 'w') do |file|
+            file.puts "#{OMS::Diag::DIAG_MIN_VERSION} xcwer23234 Release_Build"
+        end
+
+        # Forcing parsing of version file
+        OMS::Diag.SetDiagSupported(nil)
+
+        # Checking that diag is enabled
+        assert_equal(true,
+                     OMS::Diag.IsDiagEnabled(),
+                     "Diagnostic logging should be supported for when version is min version")
+    end
+
+    # Checking diagnostic support with higher omsagent version
+    def test_case_version_check_03
+        versions = ["1.3.4-128", "1.3.5-127", "1.4.4-127", "2.3.4-127"]
```

Comment:
I would also suggest adding cases like 1.4.0-1, which should evaluate as a higher version than 1.3.4-127, even though the patch and build numbers are lower.